windows-containers-multicontainer : build failed on Windows Server 2016 (Azure)

docker-compose build failed on VM Azure "Windows Server 2016 Datacenter - with Containers"

```
docker-compose -f .\docker-compose.windows.yml build
...
Step 12/16 : RUN dotnet build --framework netcoreapp1.1 .\samples\MusicStore
 ---> Running in 56d3de18fb59
Microsoft (R) Build Engine version 15.1.545.13942
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Users\ContainerAdministrator\.nuget\packages\internal.aspnetcore.sdk\1.0.1-rc2-15202\build\Git.targets(15,5): warning : Unable to determine active git branch. [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
C:\Users\ContainerAdministrator\.nuget\packages\internal.aspnetcore.sdk\1.0.1-rc2-15202\build\Git.targets(32,5): warning : Property 'CommitHash' was not set [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
StartupOpenIdConnect.cs(60,37): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
Startup.cs(42,29): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(21,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(22,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(23,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(24,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(27,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Google\TestGoogleEvents.cs(22,67): error CS0117: 'GoogleHelper' does not contain a definition for 'GetId' [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Google\TestGoogleEvents.cs(23,67): error CS0117: 'GoogleHelper' does not contain a definition for 'GetFamilyName' [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Google\TestGoogleEvents.cs(24,67): error CS0117: 'GoogleHelper' does not contain a definition for 'GetName' [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(20,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(21,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(22,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(23,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(24,99): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\StartupOpenIdConnectTesting.cs(48,37): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\StartupSocialTesting.cs(53,37): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]

Build FAILED.

C:\Users\ContainerAdministrator\.nuget\packages\internal.aspnetcore.sdk\1.0.1-rc2-15202\build\Git.targets(15,5): warning : Unable to determine active git branch. [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
C:\Users\ContainerAdministrator\.nuget\packages\internal.aspnetcore.sdk\1.0.1-rc2-15202\build\Git.targets(32,5): warning : Property 'CommitHash' was not set [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
StartupOpenIdConnect.cs(60,37): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
Startup.cs(42,29): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(21,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(22,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(23,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(24,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\MicrosoftAccount\TestMicrosoftAccountEvents.cs(27,54): error CS0103: The name 'MicrosoftAccountHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Google\TestGoogleEvents.cs(22,67): error CS0117: 'GoogleHelper' does not contain a definition for 'GetId' [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Google\TestGoogleEvents.cs(23,67): error CS0117: 'GoogleHelper' does not contain a definition for 'GetFamilyName' [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Google\TestGoogleEvents.cs(24,67): error CS0117: 'GoogleHelper' does not contain a definition for 'GetName' [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(20,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(21,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(22,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(23,54): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\Facebook\TestFacebookEvents.cs(24,99): error CS0103: The name 'FacebookHelper' does not exist in the current context [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\StartupOpenIdConnectTesting.cs(48,37): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
ForTesting\Mocks\StartupSocialTesting.cs(53,37): error CS1501: No overload for method 'UseInMemoryDatabase' takes 0 arguments [C:\MusicStore\samples\MusicStore\MusicStore.csproj]
    2 Warning(s)
    17 Error(s)

Time Elapsed 00:00:07.48
ERROR: Service 'web' failed to build: The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; dotnet build --framework netcoreapp1.1 .\samples\MusicStore' returned a non-zero code: 1
```
Docker version :
```
Client:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64
```

Docker compose version :
```
docker-compose version 1.11.1, build 7afaa436
docker-py version: 2.0.2
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
```

--- a/_posts/2017-01-20-windows-containers-multicontainer.markdown
+++ b/_posts/2017-01-20-windows-containers-multicontainer.markdown
@@ -24,9 +24,9 @@ Invoke-WebRequest https://github.com/docker/compose/releases/download/1.11.1/doc
 To try out Compose on Windows, clone a variant of the ASP.NET Core MVC MusicStore app, backed by a SQL Server Express 2016 database.
 
 ```
-git clone https://github.com/friism/Musicstore
+git clone https://github.com/docker-samples/dotnet-musicstore
 ...
-cd Musicstore
+cd dotnet-musicstore
 docker-compose -f .\docker-compose.windows.yml build
 ...
 docker-compose -f .\docker-compose.windows.yml up
@@ -36,7 +36,7 @@ docker-compose -f .\docker-compose.windows.yml up
 To access the running app from the host running the containers (for example when running on Windows 10 or if opening browser on Windows Server 2016 system running Docker engine) use the container IP and port 5000. `localhost` will not work:
 
 ```
-docker inspect -f {% raw%}{{ .NetworkSettings.IPAddress }}{% endraw %} musicstore_web_1
+docker inspect -f {% raw%}{{ .NetworkSettings.IPAddress }}{% endraw %} dotnet-musicstore_web_1
 172.21.124.54
 ```
 
@@ -46,7 +46,7 @@ If using Windows Server 2016 and accessing from outside the VM or host, simply u
 Take a closer look at the `docker-compose.windows.yml` file.
 
 ```
-version: '3'
+version: '2.1'
 services:
   db:
     image: microsoft/mssql-server-windows-express
@@ -80,14 +80,14 @@ You can find more details in the [Docker Compose documentation](https://docs.doc
   -  `web` is built from `Dockerfile.windows`. 
     - Compose passes along an environment variable which defines where the database is and how to connect to it. Notice that we can just refer to the database as `db` and Compose will allow `web` to discover the service there.
     - The port 5000 is mapped to the exposed port 5000 in the container.
-  - The two services are added to an existing network, named `nat`.
+  - The two services are added to an existing Docker network, named `nat`.
 
 Let's look at `Dockerfile.windows` to understand it a bit better.
 
 ```
-FROM microsoft/dotnet:1.0.0-preview2-windowsservercore-sdk
+FROM microsoft/dotnet:1.1-sdk-msbuild-nanoserver
 ```
-This pulls in the official microsoft .NET image based on Windows Server Core
+This pulls in the official microsoft .NET Core image based on Nano Server
 ```
 SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop';"]
 ```
@@ -95,15 +95,15 @@ This sets the shell to powershell.
 ```
 RUN set-itemproperty -path 'HKLM:\SYSTEM\CurrentControlSet\Services\Dnscache\Parameters' -Name ServerPriorityTimeLimit -Value 0 -Type DWord
 ```
-Temporary workaround for Windows DNS client weirdness
+This disables the Windows DNS cache. The cache is too aggressive, and turning it off ensures the web container always queries the DNS server running in Docker to get the address of the database container.
 ```
 RUN New-Item -Path \MusicStore\samples\MusicStore.Standalone -Type Directory
 WORKDIR MusicStore
 ```
 This creates a new directory in the container and makes it the working directory. Everything else that happens after this point will use MusicStore as the base directory.
 ```
-ADD samples/MusicStore.Standalone/project.json samples/MusicStore.Standalone/project.json
-ADD NuGet.config .
+COPY samples/MusicStore.Standalone/project.json samples/MusicStore.Standalone/project.json
+COPY NuGet.config .
 ```
 The command adds the project file (detailing NuGet package dependencies) and the NuGet config file.
 ```
@@ -112,7 +112,7 @@ RUN dotnet restore --no-cache .\samples\MusicStore.Standalone
 This pulls in the right dependencies to the project.
 
 ```
-ADD samples samples
+COPY samples samples
 RUN dotnet build .\samples\MusicStore.Standalone
 ```
 This adds the rest of the app source code to the container, and compiles the project.

